File: generated-code/codecraft/csharp/TcpReadWrite/Model/EntityType.cs
namespace TransGenTest.Model
{
    /// <summary>
    /// Entity type
    /// </summary>
    public enum EntityType
    {
        /// <summary>
        /// Wall, can be used to prevent enemy from moving through
        /// </summary>
        Wall = 0,
        /// <summary>
        /// House, used to increase population
        /// </summary>
        House = 1,
        /// <summary>
        /// Base for recruiting new builder units
        /// </summary>
        BuilderBase = 2,
        /// <summary>
        /// Builder unit can build buildings
        /// </summary>
        BuilderUnit = 3,
        /// <summary>
        /// Base for recruiting new melee units
        /// </summary>
        MeleeBase = 4,
        /// <summary>
        /// Melee unit
        /// </summary>
        MeleeUnit = 5,
        /// <summary>
        /// Base for recruiting new ranged units
        /// </summary>
        RangedBase = 6,
        /// <summary>
        /// Ranged unit
        /// </summary>
        RangedUnit = 7,
        /// <summary>
        /// Resource can be harvested
        /// </summary>
        Resource = 8,
        /// <summary>
        /// Ranged attacking building
        /// </summary>
        Turret = 9,
    }

    public static class EntityTypeHelper {
        /// <summary> Read EntityType from reader </summary>
        public static EntityType ReadFrom(System.IO.BinaryReader reader) {
            switch (reader.ReadInt32())
            {
                case 0:
                    return EntityType.Wall;
                case 1:
                    return EntityType.House;
                case 2:
                    return EntityType.BuilderBase;
                case 3:
                    return EntityType.BuilderUnit;
                case 4:
                    return EntityType.MeleeBase;
                case 5:
                    return EntityType.MeleeUnit;
                case 6:
                    return EntityType.RangedBase;
                case 7:
                    return EntityType.RangedUnit;
                case 8:
                    return EntityType.Resource;
                case 9:
                    return EntityType.Turret;
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }
    }
}

File: generated-code/php/codecraft/TcpReadWrite/model/EntityType.php
<?php

/**
 * Entity type
 */
abstract class EntityType
{
    /**
     * Wall, can be used to prevent enemy from moving through
     */
    const WALL = 0;

    /**
     * House, used to increase population
     */
    const HOUSE = 1;

    /**
     * Base for recruiting new builder units
     */
    const BUILDER_BASE = 2;

    /**
     * Builder unit can build buildings
     */
    const BUILDER_UNIT = 3;

    /**
     * Base for recruiting new melee units
     */
    const MELEE_BASE = 4;

    /**
     * Melee unit
     */
    const MELEE_UNIT = 5;

    /**
     * Base for recruiting new ranged units
     */
    const RANGED_BASE = 6;

    /**
     * Ranged unit
     */
    const RANGED_UNIT = 7;

    /**
     * Resource can be harvested
     */
    const RESOURCE = 8;

    /**
     * Ranged attacking building
     */
    const TURRET = 9;

    /**
     * Read EntityType from input stream
     */
    public static function readFrom($stream)
    {
        $result = $stream->readInt32();
        if (0 <= $result && $result < 10) {
            return $result;
        }
        throw new Exception('Unexpected tag value');
    }
}